### DISCUSSION TITLE
can’t build the project and the tests via cmake after updating to current version

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1658] [Q&A] [answered]
![image](https://github.com/axmolengine/axmol/assets/120378517/bc82a5fa-1e96-4c60-aba3-06254bcff549)
project

![image](https://github.com/axmolengine/axmol/assets/120378517/b62b05f1-fac5-46c9-859f-cd2b147a8bee)
tests

this is what happens after entering the command "cmake -S. -B build -G "Visual Studio 17 2022" -A Win32" in cmd

**Comment by halx99** [comment]
seems your git not in PATH or not installed?

**Reply by 10100010010001001010** [reply]
I agree, the second screenshot clearly shows what the problem is :) I fixed it. There was no such problem on the old version, just in case I decided to report the problem . But what about the first case?

**Reply by halx99** [reply]
try use `axmol build -p win32`

**Reply by 10100010010001001010** [reply]
![image](https://github.com/axmolengine/axmol/assets/120378517/a3ac9ab4-351a-47ea-a6ca-2a503c86c75d)

after "Configuring incomplete, errors occurred"
2 row - version MSBuild
3 row - MSBUILD: error MSB1009: file of project doesnt exist
4 row - key: 3D.vcxproj

**Reply by rh101** [reply, marked as answer]
If you searched for `ax_copy_target_dll` in both your project and the engine code (specifically `*.txt` and `*.cmake` files), you would realize that it in fact no longer exists in the engine, but you would also see where it is being used in your own source code, which is most likely in the `CMakeLists.txt` of your project.  Given that it's not something you added yourself, that should have been an indication that something has changed in the project template.

So, to save yourself time, the first thing you should always do when updating to a new version of Axmol is check if the template project has changed, and merge in any changes into your own project.

You can also generate a new project with the same name (via `axmol new`), then copy over your code to it, if that's less work for you.
